### PR TITLE
Adding Metals integration example in the docs

### PR DESCRIPTION
**Pull request by skakker**
## Chores

- [ ] linted
- [ ] tested
- [ ] documented
- [ ] changelog entry

### CODE REVIEW

**Comment by bollwyvl** [comment]
Thanks for the PR. I've not yet had the chance to review the content. I'm going to give it a whirl on binder...

[Binder](https://mybinder.org/v2/gh/skakker/jupyterlab-lsp/master?urlpath=lab%2Ftree%2Fdocs%2FConfiguring.ipynb)

As I'm tinkering, hopefully we can merge this from a _content_ perspective, and we can do a follow-on PR: move this into (and then link to) a separate notebook with just this content. If we hack it up enough to execute on binder and actually _do_ the thing, we could add this as the first of "one-off" standalone test cases on github actions.

**Comment by skakker** [comment]
Sure! That seems like a good idea.

**Comment by bollwyvl** [comment]
Sorry my review was delayed. A few things I noticed on binder:
- `java` needs to already be on path (e.g. it isn't on the binder docker image + our stuff)
  - seems obvious, but people do complain...
  - also, i don't know what i don't know, but what version is needed?
- it would be _way_ better if we got the `kernel.json` [patched upstream](https://github.com/jupyter-incubator/sparkmagic/blob/master/sparkmagic/sparkmagic/kernels/sparkkernel/kernel.json) rather than instructing folk on how to hack it
  - the next time they update, it would get overwritten, etc.

That's about as far as I got... in the meantime, I'm looking into getting `coursier` packaged for conda-forge... unversioned `curl|bash` gives me the willies, from a documentation/support perspective...

**Comment by bollwyvl** [comment]
Actually, the upstream change might be needed in the kernel message side:
https://github.com/jupyter-incubator/sparkmagic/blob/master/sparkmagic/sparkmagic/kernels/sparkkernel/sparkkernel.py#L13

**Comment by krassowski** [comment]
I opened an issue over in the sparkmagic, let's see what they think. Certainly adding file extension will be easy, not sure about changing the `language` attribute though (they seem to use some custom wiring with `session_language` instead).

**Comment by skakker** [comment]
Sorry for the delayed response. I will open a PR in the open-source project.

Even if they don't merge it, we can mention the PR here for users to try it out.

Regarding the Java version, good point. Will update the PR. The java version I have on the path is 1.8.0_212.

**Comment by skakker** [comment]
Here's the spark magic pull request: https://github.com/jupyter-incubator/sparkmagic/pull/692

**Comment by skakker** [comment]
The Spark pull request has been merged with the master

**Comment by krassowski** [comment]
@skakker now that the PR was merge, would it make sense to update the instructions here to reflect that some of the changes are no longer required?

**Comment by skakker** [comment]
@krassowski Thanks! Updated the instructions. Removed the redundant part which asked to make changes to the spark kernel.

File: docs/Configuring.ipynb [hunk 1]
```diff
@@ -188,6 +188,80 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%"
+    }
+   }
```
Comment (krassowski) [inline docs/Configuring.ipynb:264]:
```suggestion
```

File: docs/Configuring.ipynb [hunk 2]
```diff
@@ -188,6 +188,80 @@
     "[configuration files](#Configuration-Files), so that a user can fine-tune their\n",
     "available servers."
    ]
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Example: Scala Language Server (metals) integration with jupyterlab-lsp\n",
+    "\n",
+    "Step 1: Get a Scala ba:qsed kernel installed.\n",
```
Comment (krassowski) [inline docs/Configuring.ipynb:197]:
```suggestion
    "Step 1: Get a Scala based kernel installed.\n",
```

vim? ;)

File: docs/Configuring.ipynb [hunk 3]
```diff
@@ -188,6 +188,80 @@
...
+    "Metals has a coursier based installation.\n",
+    "\n",
+    "```bash\n",
+    "curl -Lo coursier https://git.io/coursier-cli && chmod +x coursier\n",
+    "./coursier bootstrap org.scalameta:metals_2.12:0.7.0 --force-fetch -o metals -f\n",
+    "```\n",
+    "\n",
+    "(Might need to use the --force-fetch flag if you are getting dependency issues.)\n",
+    "\n",
+    "Step 3: Configure the metals server in jupyterlab-lsp. Enter the following in the jupyter_notebook_config.json:\n",
```
Comment (krassowski) [inline docs/Configuring.ipynb:231]:
```suggestion
    "Step 3: Configure the metals server in jupyterlab-lsp. Enter the following in the jupyter_server_config.json:\n",
```

`jupyter_notebook_config` becomes `jupyter_server_config` for JupyterLab 3.0 / jupyter server 1.0 see https://jupyter-server.readthedocs.io/en/latest/operators/migrate-from-nbserver.html

File: docs/Configuring.ipynb [hunk 4]
```diff
@@ -188,6 +188,80 @@
...
+    "        \"argv\": [\"<$path_to_metals_server(eg:/Users/skakker/projects/jupyterlab-lsp/metals)>\"],\n",
+    "        \"languages\": [\"scala\"],\n",
+    "        \"mime_types\": [\"text/x-scala\"]\n",
+    "      }\n",
+    "    }\n",
+    "  }\n",
+    "}\n",
+    "```\n",
+    "\n",
+    "You are good to go now! Just start jupyter lab and create a notebook with either the Spark or the Scala kernel and you should be able to see the metals server initialised from the bottom left corner."
```
Comment (krassowski) [inline docs/Configuring.ipynb:248]:
```suggestion
    "You are good to go now! Just start `jupyter lab` and create a notebook with either the Spark or the Scala kernel and you should be able to see the metals server initialised from the bottom left corner."
```

or JupyterLab

**Comment by krassowski** [comment]
I think we are almost there. Please have a look on the suggestions I just made and then I will merge :)

**Comment by skakker** [comment]
The changes look good 👍  Added those to the file

**Comment by krassowski** [comment]
Thank you for your contribution @skakker :tada:

I hope we can make this Scala integration even smoother (and test it on CI) in the future, but this is a good start.

**Pull request merged**